src/flask_container_scaffold/util.py

```diff
@@ -78,5 +78,5 @@ def parse_input(logger, obj, default_return=BaseApiModel):
             parsed_args = obj.model_validate_json(json.dumps(args.to_dict()))
     except ValidationError as e:
         logger.error(f"Validation error is: {e}")
-        parsed_args = default_return(error=str(e.errors()[0]))
+        parsed_args = default_return(errors={"parse_input": e.errors()})
```

Comment:
(thought) I'm not fond of using the method name here, I don't think that's meaningful. A generic 'input_errors' may work, although my preference would be to include the field name as a key and the error as the value. I'm not sure if there's been background discussion about this already. (Example of what I have in mind below!)

(issue) The current problem for me in this design (if I understand the tests correctly!) is that a caller may get back a message that just says "Field required" with no further information. What I found with larger models is that if I have 10 fields and I get back "Field required", I couldn't see which one was missing. So in my mind, a return like this would be more helpful:

```
{
    'msg' = "XX errors were detected etc etc",
    'errors': { 
        'foo': 'Field is required', 
        'bar': 'Value must be X, Y, or Z' }
}
```

At least, this return would have helped me when I sent in the wrong input, as it is very actionable and also easy to parse. Any thoughts?

Apologies if that's beyond the scope of this patch, these are my initial thoughts as I consider how to implement it in my own project. I think I could easily override that behaviour regardless, but I wonder if this may be more generally useful.

Reply:
I have no problem with this suggestion. However, I want to mention that the pydantic errors _do_ have all that needed info, it is just a bit more detailed than what we are discussing here. The pro of that is, it may be easier to handle programatically. The con is more data comes back, so it may be harder for a human to read. Here is an example from the [docs](https://docs.pydantic.dev/2.5/errors/errors/) of one returned error object from the errors() list:
```
{
            'type': 'greater_than',
            'loc': ('gt_int',),
            'msg': 'Input should be greater than 42',
            'input': 21,
            'ctx': {'gt': 42},
            'url': 'https://errors.pydantic.dev/2/v/greater_than',
        }
```

If we choose to use this format, but want to keep our errors key (or potentially calling it details instead, which we need to discuss more outside this patch), I would also agree that naming the key for these errors based on the method name is not the path I would want. Perhaps something like 'validation_errors' for the key could be used if we went this way.

Reply:
I think that's a good schema! Thanks for the suggestion.